registry/directory/directory.go

```diff
@@ -208,8 +234,19 @@ func (dir *registryDirectory) cacheInvoker(url common.URL) {
 
 //select the protocol invokers from the directory
 func (dir *registryDirectory) List(invocation protocol.Invocation) []protocol.Invoker {
```

Comment:
This method is similar to static_directory.List(invocation). Could you refactor them?